mypyc/codegen/emitclass.py

```diff
@@ -82,11 +91,19 @@ def generate_call_wrapper(cl: ClassIR, fn: FuncIR, emitter: Emitter) -> str:
 
 def generate_slots(cl: ClassIR, table: SlotTable, emitter: Emitter) -> Dict[str, str]:
     fields = OrderedDict()  # type: Dict[str, str]
+    generated = {}  # type: Dict[str, str]
     # Sort for determinism on Python 3.5
-    for name, (slot, generator) in sorted(table.items()):
+    for name, (slot, generator) in sorted(table.items(), reverse=True):
```

Comment:
This because the implementation depends on processing `__setitem__` before `__delitem__`. This kind of hacky -- I'll add a comment in a follow-up PR, or see if this could be refactored.